### ISSUE TITLE
ACR does not expose Notary health endpoint

### ISSUE THREAD

**Issue by Starkteetje** [Azure/acr#475]
**Describe the bug**
The [ACR documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-content-trust#behind-the-scenes) mentions that "Azure Container Registry [...] is compatible with the Notary Server API". The Notary server API in the version recommended by Azure (0.6.0) [exposes a health endpoint](https://github.com/theupdateframework/notary/blob/v0.6.0/server/server.go#L234). Even when setting the ACR to be publicly accessible, this endpoint only returns 404.

**To Reproduce**
1. Create a Premium ACR instance.
2. Go to https://registryname.azurecr.io/_notary_server/health

**Expected behavior**
A 200 response.

**Any relevant environment information**
 - OS: Ubuntu
 - Datetime: Wed, 14 Oct 2020 11:49:53 GMT
 - Registry name: connyregistry (since deleted)

**Additional context**
cURL logs when connecting to ACR:
```bash
curl -v https://connyregistry.azurecr.io/_notary_server/health
*   Trying 52.178.19.3:443...
* TCP_NODELAY set
* Connected to connyregistry.azurecr.io (52.178.19.3) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=*.azurecr.io
*  start date: Oct  7 20:42:04 2020 GMT
*  expire date: Apr  7 20:42:04 2021 GMT
*  subjectAltName: host "connyregistry.azurecr.io" matched cert's "*.azurecr.io"
*  issuer: C=US; ST=Washington; L=Redmond; O=Microsoft Corporation; OU=Microsoft IT; CN=Microsoft IT TLS CA 5
*  SSL certificate verify ok.
> GET /_notary_server/health HTTP/1.1
> Host: connyregistry.azurecr.io
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Server: openresty
< Date: Wed, 14 Oct 2020 11:49:53 GMT
< Content-Type: application/octet-stream
< Content-Length: 9
< Connection: keep-alive
< Strict-Transport-Security: max-age=31536000; includeSubDomains
< 
* Connection #0 to host connyregistry.azurecr.io left intact
Not found
```

cURL logs when connecting to a local Harbor instance (Harbor also implement notary):
```bash
curl -L -v notary.harbor.domain/_notary_server/health --insecure
*   Trying 172.17.0.3:80...
* TCP_NODELAY set
* Connected to notary.harbor.domain (172.17.0.3) port 80 (#0)
> GET /_notary_server/health HTTP/1.1
> Host: notary.harbor.domain
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 308 Permanent Redirect
< Server: nginx/1.19.1
< Date: Wed, 14 Oct 2020 15:32:16 GMT
< Content-Type: text/html
< Content-Length: 171
< Connection: keep-alive
< Location: https://notary.harbor.domain/_notary_server/health
< 
* Ignoring the response-body
* Connection #0 to host notary.harbor.domain left intact
* Issue another request to this URL: 'https://notary.harbor.domain/_notary_server/health'
*   Trying 172.17.0.3:443...
* TCP_NODELAY set
* Connected to notary.harbor.domain (172.17.0.3) port 443 (#1)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=core.harbor.domain
*  start date: Oct 14 15:24:05 2020 GMT
*  expire date: Oct 14 15:24:05 2021 GMT
*  issuer: CN=harbor-ca
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x557f6bc3edb0)
> GET /_notary_server/health HTTP/2
> Host: notary.harbor.domain
> user-agent: curl/7.68.0
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200 
< server: nginx/1.19.1
< date: Wed, 14 Oct 2020 15:32:16 GMT
< content-type: application/json; charset=utf-8
< content-length: 2
< 
* Connection #1 to host notary.harbor.domain left intact
{}
```
or for Docker Hub:
```bash
curl -vL notary.docker.io/_notary_server/health 
*   Trying 3.220.75.233:80...
* TCP_NODELAY set
* Connected to notary.docker.io (3.220.75.233) port 80 (#0)
> GET /_notary_server/health HTTP/1.1
> Host: notary.docker.io
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< content-length: 0
< location: https://notary.docker.io/_notary_server/health
< 
* Connection #0 to host notary.docker.io left intact
* Issue another request to this URL: 'https://notary.docker.io/_notary_server/health'
*   Trying 3.220.75.233:443...
* TCP_NODELAY set
* Connected to notary.docker.io (3.220.75.233) port 443 (#1)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.docker.io
*  start date: May 23 00:00:00 2020 GMT
*  expire date: Jun 23 12:00:00 2021 GMT
*  subjectAltName: host "notary.docker.io" matched cert's "*.docker.io"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
> GET /_notary_server/health HTTP/1.1
> Host: notary.docker.io
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 2
< Content-Type: application/json; charset=utf-8
< Date: Wed, 14 Oct 2020 15:45:52 GMT
< Strict-Transport-Security: max-age=31536000
< 
* Connection #1 to host notary.docker.io left intact
{}
```

**Comment by Starkteetje** [comment]
Hi @SteveLasker 
thanks for the answer. We're building an [admission controller](https://github.com/sse-secure-systems/connaisseur) to verify trust data of containers being deployed in Kubernetes. Since Azure features DCT in ACR as the only one of the big clouds, we tried also integrating with it as a notary. We've now worked around it by exposing a special is_acr flag in our config and then not check for the notary health. We'd obviously like to get rid of that workaround though, so if at some point in the future you enable the health endpoint, it would be nice if you let us know :)
